src/content/developers/tutorials/short-abi/index.md

```diff
@@ -0,0 +1,632 @@
+        }
+```
+
+We could have copied the data from the call to `fallback()` (see below), but it is easier to use [Yul](https://docs.soliditylang.org/en/v0.8.12/yul.html), the assembly language of the EVM.
+
+Here we use [the CALLDATALOAD opcode](https://www.evm.codes/#35) to read bytes `startByte` to `startByte+31` into the stack.
+In general, the syntax of an opcode in Yul is `<opcode name>(<first stack value, if any>,<second stack value, if any>...)`.
+
+```solidity
+
+        _retVal = _retVal >> (256-length*8);
+```
+
+Only the most significant `length` bytes are part of the field, so we [right-shift](https://en.wikipedia.org/wiki/Logical_shift) to get rid of the other values.
+This has the added advantage of moving the value to the right of the field, so it is the value itself rather than the value times 256<sup>something</sup>.
+
+```solidity
+
+        return _retVal;
+    }
+
+
+    fallback() external {
+```
+
+When a call to a Solidity contract does not match any of the function signatures, it calls [the `fallback()` function](https://docs.soliditylang.org/en/v0.8.12/contracts.html#fallback-function) (assuming there is one).
+In the case of `CalldataInterpreter`, *any* call gets here because there are no other `external` or `public` functions.
+
+
+
+```solidity
+        uint _func;
+
+        _func = calldataVal(0, 1);
+```
+
+Read the first byte of the calldata, which tells us the function.
+There are two reasons why a function would not be available here:
+
+1. Functions that are `pure` or `view` don't change the state and don't cost gas (when called off-chain).
+   It makes no sense to try to reduce their gas cost.
+   
+2. Functions that rely on [`msg.sender`](https://docs.soliditylang.org/en/v0.8.12/units-and-global-variables.html#block-and-transaction-properties).
+   The value of `msg.sender` is going to be `CalldataInterpreter`'s address, not the caller.
+   
+Unfortunately, [looking at the ERC-20 specifications](https://eips.ethereum.org/EIPS/eip-20), this leaves only one function, `transfer`.
+This leaves us with only two functions: `transfer` (because we can call `transferFrom`) and `faucet` (because we can transfer the tokens back to whever called us).
+
+
+```solidity
+
+        // Call the state changing methods of token using
+        // information from the calldata
+
+        // faucet
+        if (_func == 1) {
+```
+
+A call to `faucet()`, which doesn't have parameters.
+
+```solidity
+            token.faucet();
+            token.transfer(msg.sender,
+                token.balanceOf(address(this)));
+        }
+```
+
+After we call `token.faucet()` we get tokens. However, as the proxy contract, we do not **need** tokens. 
+The EOA (externally owned account) or contract that called us does.
+So we transfer all of our tokens to whoever called us.
+
+
+```solidity
+        // transfer (assume we have an allowance for it)
+        if (_func == 2) {
+```
+
+Transfer tokens.
```

Comment:
Is this a heading?

Reply:
No, I just put each sentence in its own line. But I decided to delete this sentence anyway.